### PR TITLE
refactor: command line and masterdata processing

### PR DESCRIPTION
**Pull request by ml019**
## Description
- replace direct access to the `commandLineOption` global variable with accessor routines
- accessor routines obtain their input via the input state associated with the current input source
- entrances can now contribute to the input seeders used in order to modify the state prior to the entrance being invoked. This is mainly to permit adjustment to the command line options prior to their validation as part of invoking the entrance.
- populate masterdata via the input source process

The input source mechanism has also been expanded to support multiple input pipelines, mainly to permit handling of output modification depending on the input source requested. A subsequent change will complete the use of this mechanism - at present output handling has been left unaffected.

The previous `mock` stage has been renamed to `fixture` but the input source has been left as `mock`. Given outputs in some input sources are also mocked, `fixture` seemed a more specific term describing the purpose of the stage.

## Motivation and Context
Part of the transition to the use of input seeders.

    
## How Has This Been Tested?
Local template creation

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions
These PRs need to be released at the same time to align with the changes in this PR.

- [x] AWS support - https://github.com/hamlet-io/engine-plugin-aws/pull/255
- [x] Azure support -  https://github.com/hamlet-io/engine-plugin-azure/pull/238

## Checklist:

- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] None of the above.

### CODE REVIEW

**Comment by roleyfoley** [comment]
@ml019 Is this PR now ready for review?

**Comment by roleyfoley** [comment]
Will be replaced with a new one

File: providers/shared/layers/layer.ftl [hunk 1]
```diff
@@ -18,6 +18,9 @@
 [#assign SEGMENT_LAYER_TYPE = "Segment" ]
 [#assign SEGMENT_LAYER_REFERENCE_TYPE = "Segments" ]
 
+[#assign REGION_LAYER_TYPE = "Segment" ]
+[#assign REGION_LAYER_REFERENCE_TYPE = "Segments" ]
```
Comment (roleyfoley) [inline providers/shared/layers/layer.ftl:22]: Should these values be Region and Regions?

File: engine/inputdata/commandLineOptions.ftl [hunk 2]
```diff
@@ -22,15 +22,109 @@
...
+[/#function]
+
+[#function getCompositeDefinitions]
+    [#return (getCommandLineOptions().Composites.Definitions)!{} ]
+[/#function]
+
+[#function getCompositeStackOutputs]
+    [#return (getCommandLineOptions().Composites.StackOutputs)![] ]
+[/#function]
+
```
Comment (roleyfoley) [inline engine/inputdata/commandLineOptions.ftl:126]: Should these all be namespaced to getCLO  instead of just a couple?

**Review by roleyfoley — CHANGES_REQUESTED** [review]

**Comment by RossMurr4y** [comment]
Performed local testing and schema generation against this and matching provider PRs and all were successful.

**Review by roleyfoley — APPROVED** [review]

**Pull request merged**